### PR TITLE
core-services/prow/02_config/_boskos: Shard AWS, Azure, and GCP by region

### PR DESCRIPTION
**Pull request by wking**
With some guessing about what we support.  No consumers yet, but consumer rehearsals [run against the production Boskos server][1], so we need to land these before we can consume them.

[1]: https://github.com/openshift/release/pull/12584#issuecomment-705890665

### CODE REVIEW

File: core-services/prow/02_config/_boskos.yaml
Comment (alvaroaleman) [inline core-services/prow/02_config/_boskos.yaml:4]: These are the names of the resulting crds, so this will result in exactly one. The boskos resources have a `userdata` field, maybe we can leverage that?

Comment (alvaroaleman) [reply core-services/prow/02_config/_boskos.yaml:4]: Right now we expose the name of the resource via an env var, wie could expose the userdata via another env var.

Comment (wking) [reply core-services/prow/02_config/_boskos.yaml:4]: I don't understand how Boskos works.  @stevekuznetsov recommended this "repeat entries in `names`" approach back on 2020-09-10.  He also mentioned userdata and linked [here][1].  I don't really care how I'm supposed to do it, I just want a way to manage per-region leases.  If this approach isn't right, can you spell out what you want in more detail?

[1]: https://github.com/kubernetes-sigs/boskos#adding-userdata-to-a-resource

Comment (wking) [reply core-services/prow/02_config/_boskos.yaml]: Ok, rerolled to enumerate the repeat entries with 61e9a5ae65 -> ef394fae36.

File: core-services/prow/02_config/_boskos.yaml
Comment (alvaroaleman) [inline core-services/prow/02_config/_boskos.yaml:256]: Nit: Experience teaches that if you gurantee a certain number of digits, ppl will assume this is always true: https://www.hyrumslaw.com/ 

Hence, its usually better to just count up. WDYT?

Comment (wking) [reply core-services/prow/02_config/_boskos.yaml]: openshift/ci-tools#1306 hides this from consumers, and with 0ec8261ff8 I have a guard here to keep folks from accidentally tripping over the ci-tools logic.

**Review by alvaroaleman — APPROVED** [review]
/hold
until the corresponding ci-tools pr is merged and rolled out

**Comment by wking** [comment]
[ci-tools postsubmit][1]:

```
2020/10/14 21:34:40 Resolved source https://github.com/openshift/ci-tools to master@c4a0d926
...
2020/10/14 21:51:09 Promoting tags to ci/${component}:latest: ... ci-operator, ...
```

Looks like the local consumers are all `image: ci-operator:latest`, so that should be enough to pull the hold, right?

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-openshift-ci-tools-master-images/1316491028840058880#1:build-log.txt%3A151

**Comment by wking** [comment]
4h since my last comment.  Surely it has made it over by now, and even if it hasn't, nobody is consuming the `--` entries yet.

/hold cancel

**Pull request merged**
